src/System Application/App/Agent/Setup/AgentImpl.Codeunit.al

```diff
@@ -425,6 +426,41 @@ codeunit 4301 "Agent Impl."
         Hyperlink(NoAgentsAvailableNotificationLearnMoreUrlLbl);
     end;
 
+    local procedure GenerateUniqueUserName(AgentUserName: Code[50]): Code[50]
+    var
+        User: Record User;
+        UniqueUserName: Text[50];
+        AgentNamePrefix: Text[50];
+        NumberOfAgentDigits: Integer;
+        MaximumPrefixLength: Integer;
+        AgentNumberSeparatorTok: Label '-', Locked = true;
+    begin
+        // Check if the user name is already unique
+        User.Ascending(false);
```

Comment:
No reason for `User.Ascending(false)`, we are filtering on `"User Name"` and check `IsEmpty()`